docs/Setup/ca-setup.md

```diff
@@ -0,0 +1,182 @@
+
+### TLS Certificate Authority
+
+In addition to _enrollment certificates_ and _transaction certificates_, users will need _TLS certificates_ to secure their communication channels. _TLS certificates_ can be requested from the _TLS certificate authority_ (TLSCA).
+
+## Configuration
+
+All CA services are provided by a single process, which can be configured by setting parameters in the CA configuration file `membersrvc.yaml`, which is located in the same directory as the CA binary. More specifically, the following parameters can be set:
+
+- `server.gomaxprocs`: limits the number of operating system threads used by the CA.
+- `server.rootpath`: the root path of the directory where the CA stores its state.
+- `server.cadir`: the name of the directory where the CA stores its state.
+- `server.port`: the port at which all CA services listen (multiplexing of services over the same port is provided by [GRPC](http://www.grpc.io)).
+
+Furthermore, logging levels can be enabled/disabled by adjusting the following settings:
+
+- `logging.trace` (off by default, useful for debugging the code only)
+- `logging.info`
+- `logging.warning`
+- `logging.error`
+- `logging.panic`
+
+Alternatively, these fields can be set via environment variables, which---if set---have precedence over entries in the yaml file. The corresponding environment variables are named as follows:
+
+```
+    MEMBERSRVC_CA_SERVER_GOMAXPROCS
+    MEMBERSRVC_CA_SERVER_ROOTPATH
+    MEMBERSRVC_CA_SERVER_CADIR
+    MEMBERSRVC_CA_SERVER_PORT
+```
+
+In addition, the CA may be preloaded with registered users, where each user's name, roles, and password are specified:
+
+```
+    eca:
+    	users:
+    		alice: 2 DRJ20pEql15a
+    		bob: 4 7avZQLwcUe9q
+```
+The role value is simply a bitmask of the following:
+```
+    CLIENT = 1;
+    PEER = 2;
+    VALIDATOR = 4;
+    AUDITOR = 8;
+```
+
+For example, a peer that is also a validator would have a role value of 6.
+
+When the CA is started for the first time, it will generate all of its required state (e.g., internal databases, CA certificates, blockchain keys, etc.) and writes this state to the directory given in its configuration. The certificates for the CA services (i.e., for the ECA, TCA, and TLSCA) are self-signed as the current default. If those certificates shall be signed by some root CA, this can be done manually by using the `*.priv` and `*.pub` private and public keys in the CA state directory, and replacing the self-signed `*.cert` certificates with root-signed ones. The next time the CA is launched, it will read and use those root-signed certificates.
```

Comment:
@joshhus sorry, thought you were done... @JonathanLevi plans to take on ca-setup.md. Please feel free to submit PRs for any other changes.